### PR TITLE
RFC: large refactoring of how the package works

### PR DESCRIPTION
**Pull request by odow**
See also https://github.com/lanl-ansi/MINLPTests.jl/pull/6

### CODE REVIEW

**Comment by ccoffrin** [comment]
general direction looks good to me.

File: test/runtests.jl [hunk 1]
```diff
@@ -1,48 +1,37 @@
...
-end
-
-for solver in minlp_solvers
-    @testset "$(string(typeof(solver))) MINLP Tests" begin
-        MINLPTests.test_nlp_mi(solver)
+    @testset "$(string(typeof(solver))): nlp_mi" for solver in MINLP_SOLVERS
+        MINLPTests.test_nlp_mi(solver, exclude = [
+            "005_011",  # Uses the function `\`
+            "003_013",  # Bug in Juniper - handling of expression graph?
+            "006_010"   # Bug in Juniper - handling of user-defined functions.
```
Comment (odow) [inline test/runtests.jl]: @ccoffrin are these expected failures for Juniper?

**Comment by odow** [comment]
This is good to merge on my end, then I can get on with rebasing the `moi` branch.

File: test/runtests.jl (hunk 1 above)
Comment (ccoffrin) [reply test/runtests.jl]: Yes, they were failing when I built the dataset, https://github.com/lanl-ansi/MINLPTests.jl/blob/master/src/nlp-mi/tests.jl

**Comment by ccoffrin** [comment]
Merge at your convenience.

**Pull request merged**